maint/towncrier_rst_validator.py

```diff
@@ -0,0 +1,89 @@
+from subprocess import STDOUT, check_output
+import argparse
+import os
+
+parser=argparse.ArgumentParser()
+
+parser.add_argument("--pull_request_id", required=True, type=int)
+
+args=parser.parse_args()
+
+rst_dir = "docs/upcoming_changes"
+types_of_changes = ["highlight",
+                    "np_support",
+                    "deprecation",
+                    "expired",
+                    "compatibility",
+                    "cuda",
+                    "new_feature",
+                    "improvement",
+                    "performance",
+                    "change",
+                    "doc",
+                    "infrastructure",
+                    "bug_fix"]
+
+def list_rst_filename() -> str:
+    output = check_output(
+        ["git", "diff", "--name-only", "origin/main"],
+        encoding="utf-8",
+        stderr=STDOUT,
+    )
+
+    all_files = output.strip().splitlines()
+    rst_file = [file for file in all_files if 
+                file.startswith(rst_dir + "/" + str(args.pull_request_id) + ".")]
+    assert len(rst_file) == 1, "There must be a relevant .rst file added in the PR." + \
+        "Filename must start with the respective Pull Request ID" + \
+        " (eg 1234. for PR #1234) followed by a . character"
```

Comment:
Suggestion:

- print content of `rst_file`
- Error message is confusing when no file is found. (I thought the script will let me run the validator on any correctly named .rst file but it turns out only modified files is checked. Btw, having a mode that can check any file even if they are not in the diff will be handy).

Reply:
Have added such a mode (and resolved above comments) in b42af38